aiidalab_widgets_base/structures.py

```diff
@@ -259,13 +259,27 @@ def _observe_structure_node(self, change):
             self.structure_description.value = ''
             self.structure_description.disabled = False
 
+    def _validate_and_fix_ase_cell(self, ase_structure, vacuum_ang=10.0):
+        """
+        Checks if the ase Atoms object has a cell set,
+        otherwise sets it to bounding box plus specified "vacuum" space
+        """
+        cell = ase_structure.cell
+
+        if (np.linalg.norm(cell[0]) < 0.1 or np.linalg.norm(cell[1]) < 0.1 or np.linalg.norm(cell[2]) < 0.1):
+            # if any of the cell vectors is too short, consider it faulty
+            # set cell as bounding box + vacuum_ang
+            bbox = np.ptp(ase_structure.positions, axis=0)
+            ase_structure.cell = bbox + vacuum_ang
```

Comment:
```suggestion
            bbox = np.ptp(ase_structure.positions, axis=0)
            new_structure = ase_structure.copy()
            new_structure.cell = bbox + vacuum_ang
            return new_structure
        return ase_structure
```

Reply:
The point here is that I don't want to modify the original structure object with some unexpected consequences.